### PR TITLE
Rename "normalize" to "sanitize"

### PR DESCRIPTION
**Pull request by breedx-splk**
As suggested in #2065, it makes more sense to call this "sanitization" instead of "normalization", so this is just a rename for clarity.

It doesn't tackle the `normalizeQuery()` method on the DatabaseClientTracer, because that appears to be wholly separate concern (more like "extractQuery()" or something).

### CODE REVIEW

File: javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/sanitizer/SqlSanitizerTest.groovy [hunk 1]
```diff
@@ -116,9 +116,9 @@ class SqlNormalizerTest extends Specification {
     for (int i = 0; i < 10000; i++) {
       s.append("SELECT * FROM TABLE WHERE FIELD = 1234 AND ")
     }
-    String normalized = SqlNormalizer.normalize(s.toString())
+    String normalized = SqlSanitizer.sanitize(s.toString())
```
Comment (anuraaga) [inline javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/sanitizer/SqlSanitizerTest.groovy:119]: sanitized

**Review by anuraaga — APPROVED** [review]
Thanks!

**Review by trask — APPROVED** [review]
👍

**Review by mateuszrzeszutek — APPROVED** [review]

File: javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/sanitizer/SqlSanitizerTest.groovy (hunk 1 above)
Comment (breedx-splk) [reply javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/sanitizer/SqlSanitizerTest.groovy]: solid.

**Pull request merged**